WordPress/src/main/java/org/wordpress/android/ui/notifications/services/NotificationsPendingDraftsService.java

```diff
@@ -0,0 +1,229 @@
+                    }
+
+                    //check the size and build the notification accordingly
+                    long daysInDraft = 0;
+                    if (draftPostsOlderThan3Days.size() == 1) {
+                        Post post = draftPostsOlderThan3Days.get(0);
+                        daysInDraft = (now - post.getDateLastUpdated()) / ONE_DAY;
+                        long postId = post.getLocalTablePostId();
+                        boolean isPage = post.isPage();
+
+                        //only notify the user if the last time they have been notified about this particular post was at least
+                        //30 minutes ago, but let's not do it constantly each time the app comes to the foreground
+                        if ((now - post.getDateLastNotified()) > MINIMUM_ELAPSED_TIME_BEFORE_REPEATING_NOTIFICATION) {
+                            post.setDateLastNotified(now);
+                            if (daysInDraft < MAX_DAYS_TO_COUNT_DAYS) {
+                                buildSinglePendingDraftNotification(daysInDraft, postId, isPage);
+                            } else {
+                                //if it's been more than MAX_DAYS_TO_COUNT_DAYS days, or if we don't know (i.e. value for lastUpdated
+                                //is zero) then just show a generic message
+                                buildSinglePendingDraftNotification(postId, isPage);
+                            }
+                            WordPress.wpDB.updatePost(post);
+                        }
+
+                    } else if (draftPostsOlderThan3Days.size() > 1) {
+                        long longestLivingDraft = 0;
+                        boolean onlyPagesFound = true;
+                        for (Post post : draftPostsOlderThan3Days) {
+                            //update each post dateLastNotified field to now
+                            if ((now - post.getDateLastNotified()) > MINIMUM_ELAPSED_TIME_BEFORE_REPEATING_NOTIFICATION) {
+                                if (post.getDateLastNotified() > longestLivingDraft) {
+                                    longestLivingDraft = post.getDateLastNotified();
+                                    if (!post.isPage()) {
+                                        onlyPagesFound = false;
+                                    }
+                                }
+                                post.setDateLastNotified(now);
+                                WordPress.wpDB.updatePost(post);
+                            }
+                        }
+
+                        //if there was at least one notification that exceeded the minimum elapsed time to repeat the notif,
+                        //then show the notification again
+                        if (longestLivingDraft > 0) {
+                            buildPendingDraftsNotification(draftPostsOlderThan3Days.size(), onlyPagesFound);
+                        }
+                    }
+
+                    completed();
```

Comment:
This should be move outside the `if` condition `if (draftPosts != null && draftPosts.size() > 0)`